packages/analytics/src/get-config.test.ts

```diff
@@ -0,0 +1,192 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { SinonStub, stub, useFakeTimers } from 'sinon';
+import '../testing/setup';
+import { fetchDynamicConfig, fetchDynamicConfigWithRetry } from './get-config';
+import { DYNAMIC_CONFIG_URL } from './constants';
+import { getFakeApp } from '../testing/get-fake-firebase-services';
+import { DynamicConfig } from '@firebase/analytics-types';
+import { AnalyticsError } from './errors';
+
+const fakeMeasurementId = 'abcd-efgh-ijkl';
+const fakeAppId = 'abcdefgh12345:23405';
+const fakeAppParams = { appId: fakeAppId, apiKey: 'AAbbCCdd12345' };
+const fakeUrl = DYNAMIC_CONFIG_URL.replace('{app-id}', fakeAppId);
+const successObject = { measurementId: fakeMeasurementId, appId: fakeAppId };
+let fetchStub: SinonStub;
+
+function stubFetch(status: number, body: { [key: string]: any }): void {
+  fetchStub = stub(window, 'fetch');
+  const mockResponse = new window.Response(JSON.stringify(body), {
+    status
+  });
+  fetchStub.returns(Promise.resolve(mockResponse));
+}
+
+describe('Dynamic Config Fetch Functions', () => {
+  describe('fetchDynamicConfig() - no retry', () => {
+    it('successfully request and receives dynamic config JSON data', async () => {
+      stubFetch(200, successObject);
+      const app = getFakeApp(fakeAppParams);
+      const config: DynamicConfig = await fetchDynamicConfig(app);
+      expect(fetchStub.args[0][0]).to.equal(fakeUrl);
+      expect(fetchStub.args[0][1].headers.get('x-goog-api-key')).to.equal(
+        fakeAppParams.apiKey
+      );
+      expect(config.appId).to.equal(fakeAppId);
+      expect(config.measurementId).to.equal(fakeMeasurementId);
+    });
+    it('throws error on failed response', async () => {
+      stubFetch(500, {
+        error: {
+          /* no message */
+        }
+      });
+      const app = getFakeApp(fakeAppParams);
+      let error;
+      try {
+        await fetchDynamicConfig(app);
+      } catch (e) {
+        error = e;
+      }
+      expect(error.message).to.include(AnalyticsError.CONFIG_FETCH_FAILED);
```

Comment:
I think in that case if the catch block isn't reached, the test will "pass", since it does not hit the `expect` statement. I tried a more compact form, `to.be.rejectedWith`.